hazelcast/src/main/java/com/hazelcast/spi/impl/sequence/CallIdSequenceWithBackpressure.java

```diff
@@ -94,13 +94,20 @@ private boolean hasSpace() {
         return longs.get(INDEX_HEAD) - longs.get(INDEX_TAIL) < maxConcurrentInvocations;
     }
 
-    private void waitForSpace() throws TimeoutException {
-        long deadline = System.nanoTime() + backoffTimeoutNanos;
+    private void waitForSpace() {
+        if (backoffTimeoutNanos <= 0) {
+            throw new HazelcastOverloadException(String.format(
+                    "Maximum invocation count reached." + " maxConcurrentInvocations = %d, backoffTimeout = %d msecs",
+                    maxConcurrentInvocations, NANOSECONDS.toMillis(backoffTimeoutNanos)));
+        }
+
+        long start = System.nanoTime();
         for (long idleCount = 0; ; idleCount++) {
-            if (System.nanoTime() >= deadline) {
-                throw new TimeoutException(String.format("Timed out trying to acquire another call ID."
-                        + " maxConcurrentInvocations = %d, backoffTimeout = %d", maxConcurrentInvocations,
-                        NANOSECONDS.toMillis(backoffTimeoutNanos)));
+            long now = System.nanoTime();
+            if (now - start > backoffTimeoutNanos) {
+                throw new HazelcastOverloadException(String.format("Timed out trying to acquire another call ID."
```

Comment:
Printing of now/start makes no sense in case of nano time. Nano time is not related to epoch time.

Reply:
I think `start` is fine, since this also fits a relative time measurement. The `now` could be avoided by using `long elapsedNanos = System.nanoTime() - start;` and then using this.

Reply:
OK, I am changing it to print the elapsed time.